From 8aa84e63582c4682212f35b283094e81dfaf7748 Mon Sep 17 00:00:00 2001
From: Andre Amegah <andrex.amegah@intel.com>
Date: Thu, 2 May 2013 15:35:14 +0300
Subject: [PORT FROM MAIN] Telephony: Fix issue with image data EF read.

BZ: 100137

Reading of icon data from the image instance
data EF file fails. The path parameter passed
for fetching the actual image data was
wrong. Provided path was under DF phonebook whereas
the requested file should be under DF Graphics.

This patch fixes issue by providing the right file
path.

Change-Id: I4109ff44c6458b5fe98994593d1b10d38e0e7b39
Orig-Change-Id: Ifd260a0d606331fd66e7568f7ef6452f99cd8531
Signed-off-by: Andre Amegah <andrex.amegah@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/uicc/IccFileHandler.java    |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/IccFileHandler.java b/src/java/com/android/internal/telephony/uicc/IccFileHandler.java
index 47641f4..791c416 100644
--- a/src/java/com/android/internal/telephony/uicc/IccFileHandler.java
+++ b/src/java/com/android/internal/telephony/uicc/IccFileHandler.java
@@ -260,7 +260,7 @@ public abstract class IccFileHandler extends Handler implements IccConstants {
                 onLoaded);
 
         logd("IccFileHandler: loadEFImgTransparent fileid = " + fileid
-                + " filePath = " + getEFPath(fileid) + " highOffset = " + highOffset
+                + " filePath = " + getEFPath(IccConstants.EF_IMG) + " highOffset = " + highOffset
                 + " lowOffset = " + lowOffset + " length = " + length);
         /*
          * Per TS 31.102, for displaying of Icon, under
@@ -268,7 +268,7 @@ public abstract class IccFileHandler extends Handler implements IccConstants {
          * are present. The possible image file identifiers (EF instance) for
          * EF img ( 4F20, linear fixed file) are : 4F01 ... 4F05.
          */
-        mCi.iccIOForApp(COMMAND_READ_BINARY, fileid, getEFPath(fileid),
+        mCi.iccIOForApp(COMMAND_READ_BINARY, fileid, getEFPath(IccConstants.EF_IMG),
                 highOffset, lowOffset, length, null, null, mAid, response);
     }
 
-- 
1.7.4.1

